### PR TITLE
Add bounds/controlPointBounds properties

### PR DESCRIPTION
**Pull request by khaledhosny**
For compatibility with defcon.

### CODE REVIEW

File: src/ufoLib2/objects/layer.py [hunk 1]
```diff
@@ -203,6 +209,44 @@ def name(self) -> str:
...
+        """
+        bounds = None
+        for glyph in self:
+            glyphBounds = glyph.getBounds(self)
+            if glyphBounds is None:
+                continue
+            if bounds is None:
+                bounds = glyphBounds
+            else:
+                bounds = unionBounds(bounds, glyphBounds)
```
Comment (khaledhosny) [inline src/ufoLib2/objects/layer.py:227]: It is probably too late for me to think properly, but I have absolutely no idea why mypy thinks this line is unreachable.

File: src/ufoLib2/objects/contour.py [hunk 2]
```diff
@@ -147,7 +147,16 @@ def getControlBounds(
         """
         return getControlBounds(self, layer)
 
-    # XXX: Add property controlPointBounds (defcon compat API)?
+    @property
+    def controlPointBounds(self) -> Optional[BoundingBox]:
+        """Returns the (xMin, yMin, xMax, yMax) bounding box of the glyph,
+        taking only the control points into account.
+
+        Gives inaccurate results with extruding curvatures.
```
Comment (anthrotype) [inline src/ufoLib2/objects/contour.py:155]: What do you mean? what's extruding curvatures?

Comment (anthrotype) [reply src/ufoLib2/objects/contour.py:155]: Ah maybe you mean curves that don't have control points at their extremes. But I wouldn't characterize the result from the ControlBoundsPen as inaccurate. They are what they are, and curve is always contained within the control points' bounds (by the convex hull property), so they are correct. If one wants "tighter" bounds one can use the other `bounds` property.

Comment (khaledhosny) [reply src/ufoLib2/objects/contour.py:155]: No idea 😄 I copied the doc string of the corresponding Contour functions.

Comment (anthrotype) [reply src/ufoLib2/objects/contour.py:155]: ah :) so the original text was incorrect /cc @madig

Comment (madig) [reply src/ufoLib2/objects/contour.py:155]: i think i copied it from defcon, or maybe i didn't

Comment (anthrotype) [reply src/ufoLib2/objects/contour.py:155]: ok let's just remove the sentence

Comment (khaledhosny) [reply src/ufoLib2/objects/contour.py]: Done

**Comment by khaledhosny** [comment]
> Thanks! Could you add a test as well?

I added a font test, which I guess calls the other code. I couldn’t figure out how to add more tests.

**Pull request merged**